tests/Availability/AvailabilityTest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Prophecy\PhpUnit\Tests\Availability;
+
+use PHPUnit\Framework\TestCase;
+use Prophecy\PhpUnit\ProphecyTrait;
+
+/**
+ * Testing that the autoloading of the empty `ProphecyTrait` trait for PHPUnit 4.x - 9.1 works correctly.
+ *
+ * {@internal The code in this file must be PHP cross-version compatible for PHP 5.4 - current!}
+ *
+ * @coversNothing
+ */
+final class AvailabilityTest extends TestCase
```

Comment:
this tes is not implemented as a proper testcase for the availability. It won't produce a testsuite failure if the trait is not available, but a fatal error in the test loading.

The availability test should look like that:

```php
final class AvailabilityTest extends TestCase
{
    public function testTraitIsAvailable()
    {
        $this->assertTrue(trait_exists(ProphecyTrait::class);
    }
}
```

Reply:
I implemented it as an availability test for the `prophesize()` _method_, not for the _trait_, though that is indirectly also tested as the trait is _used_ by this test, so if it would not be available, the test would error out on a fatal.

Maybe I should adjust the test description in the docblock to make that clearer?

Reply:
An alternative implementation of the test would IMO look like this:
```php
final class AvailabilityTest extends TestCase
{
    use ProphecyTrait;

    public function testTraitIsAvailable()
    {
        $this->assertTrue(method_exists($this, 'prophesize'));
    }
}
```

Reply:
this is still not the right thing. You cannot use the current test case as the subject being tested. This does not play well with failure reporting.

The case of checking that the `prophesize` method exist means that we should indeed have a second assertion using a fixture class (as done in the `ProphecyTraitTest`) to check for the method existence.
but to me, the `trait_exists` existence check (*before* using the fixture class using that trait) still makes sense, to have a proper failure if the trait does not exist properly (due to a mistake in loading it) rather than a fatal error.